aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -30,12 +29,6 @@
 
 public class AwsClientFactories {
 
-  /**
-   * The implementation class of {@link AwsClientFactory} to customize AWS client configurations.
-   * If set, all AWS clients will be configured by the specified class before initialization.
-   */
-  public static final String CLIENT_FACTORY_CONFIG_KEY = "client.factory";
-
   private static final SdkHttpClient HTTP_CLIENT_DEFAULT = UrlConnectionHttpClient.create();
```

Comment:
Unrelated to this change: 

* Did we consider using SdkAsyncHttpClient such as [NettyNioAsyncHttpClient](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.html) for non blocking IO.

* Should we have different HTTP clients across different services to avoid contention for the shared connection pool and also have flexibility to configure the http client settings based on the service?

* Should we use async service clients like S3AsyncClient etc

Reply:
Yes we considered about using Netty async implementation and async AWS client, especially for s3. There was some discussion about that in #1767, and some people mentioned that Netty implementation has some unresolved bug. Since the existing solution using java future is good enough, we just chose that route. 

I will do some testing to see if it can improve S3 FileIO performance, and if it works I will raise another PR. The API for client factory can always be updated for async clients.